ntbs-service/Pages/Notifications/Edit/SocialRiskFactors.cshtml.cs

```diff
@@ -0,0 +1,98 @@
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using ntbs_service.Models;
+using ntbs_service.Models.Enums;
+using ntbs_service.Pages;
+using ntbs_service.Services;
+using System.Linq;
+
+namespace ntbs_service.Pages_Notifications
+{
+    public class SocialRiskFactorsModel : ValidationModel
+    {
+        private readonly INotificationService service;
+        private readonly NtbsContext context;
+        
+        public List<Status> StatusList { get; set; }
+
+        [BindProperty]
+        public SocialRiskFactors SocialRiskFactors { get; set; }
+
+        [BindProperty]
+        public int NotificationId { get; set; }
+        
+
+        public SocialRiskFactorsModel(INotificationService service, NtbsContext context)
+        {
+            this.service = service;
+            this.context = context;
+        }
+
+        public async Task<IActionResult> OnGetAsync(int? id)
+        {
+            if (id == null)
+            {
+                return NotFound();
+            }
+
+            var notification = await service.GetNotificationWithSocialRisksAsync(id);
+            NotificationId = notification.NotificationId;
+            SocialRiskFactors = notification.SocialRiskFactors;
+
+            StatusList = Enum.GetValues(typeof(Status)).Cast<Status>().ToList();
+
+            if (SocialRiskFactors == null)
+            {
+                return NotFound();
+            }
+
+            return Page();
+        }
+
+        public IActionResult OnGetPartial() => Partial("_RiskFactorPartial");
+
+        public async Task<IActionResult> OnPostPreviousPageAsync(int? NotificationId)
+        {
+            var notification = await service.GetNotificationWithSocialRisksAsync(NotificationId);

```

Comment:
Since, due to your bugfix, other places clear "flagged" data here, do we want to follow the same pattern?